internal/backends/sqlite/backend.go

```diff
@@ -29,37 +31,79 @@ import (
 // backend implements backends.Backend interface.
 type backend struct {
 	r   *metadata.Registry
-	dir string
+	uri *url.URL
 }
 
 // NewBackendParams represents the parameters of NewBackend function.
 type NewBackendParams struct {
-	Dir string
+	URI string
 	L   *zap.Logger
 }
 
 // NewBackend creates a new SQLite backend.
 func NewBackend(params *NewBackendParams) (backends.Backend, error) {
-	fi, err := os.Stat(params.Dir)
+	uri, err := validateURI(params.URI)
 	if err != nil {
-		return nil, lazyerrors.Errorf("%q should be an existing directory: %w", params.Dir, err)
+		return nil, err
 	}
 
-	if !fi.IsDir() {
-		return nil, lazyerrors.Errorf("%q should be an existing directory", params.Dir)
-	}
-
-	r, err := metadata.NewRegistry(params.Dir, params.L.Named("metadata"))
+	r, err := metadata.NewRegistry(uri, params.L.Named("metadata"))
 	if err != nil {
 		return nil, lazyerrors.Error(err)
 	}
 
 	return backends.BackendContract(&backend{
 		r:   r,
-		dir: params.Dir,
+		uri: uri,
 	}), nil
 }
 
+// validateURI checks given URI value and returns parsed URL.
+// URI should contain 'file' scheme and point to an existing directory.
+// Path should end with '/'. Authority should be empty.
```

Comment:
Authority could also be absent: empty – `file://<empty>/`, absent – `file:/`. See https://en.wikipedia.org/wiki/Uniform_Resource_Identifier#Syntax

Reply:
Mentioned that it could be absent.